src/user-lists/user-list.service.ts

```diff
@@ -309,7 +309,16 @@ export class UserListService {
     queryBuilder.orderBy("full_name", orderBy);
     queryBuilder.offset(pageOptionsDto.skip).limit(pageOptionsDto.limit);
 
-    const [itemCount, entities] = await Promise.all([queryBuilder.getCount(), queryBuilder.getMany()]);
+    const subQuery = this.userHighlightRepository.manager
+      .createQueryBuilder()
+      .from(`(${queryBuilder.getQuery()})`, "subquery_for_count")
+      .setParameters(queryBuilder.getParameters())
+      .select("count(full_name)");
```

Comment:
This works well. But, for convention, we've moved away from sub queries since they have caused problems in the past: it's almost the same as what you have:

```suggestion
     const subQuery = this.userHighlightRepository.manager
       .createQueryBuilder()
       .from(`(${queryBuilder.getQuery()})`, "subquery_for_count")
       .addCommonTableExpression(queryBuilder, "CTE")
       .setParameters(queryBuilder.getParameters())
       .select("count(full_name)");
       .select("count(full_name)")
       .from("CTE", "CTE");
```

Reply:
Thanks for the clarification

Reply:
I tested the above common table expression diff, we should be good to commit that chunk and merge this 👍🏼

Reply:
```typescript
     .createQueryBuilder()
       .from(`(${queryBuilder.getQuery()})`, "subquery_for_count")
       .addCommonTableExpression(queryBuilder, "CTE")
       .setParameters(queryBuilder.getParameters())
       .select("count(full_name)");
       .select("count(full_name)")
       .from("CTE", "CTE");
```
This expression causes a break... removing line 322, 323, and 3324 appears to work as expected.